api/v1alpha1/nicclusterpolicy_types.go

```diff
@@ -101,16 +102,24 @@ type SecondaryNetworkSpec struct {
 	IpamPlugin *ImageSpec `json:"ipamPlugin,omitempty"`
 }
 
+type ClusterNodeSelectorSpec struct {
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	ClusterNodeSelector map[string]string `json:"nodeSelector,omitempty"`
+}
+
 // NicClusterPolicySpec defines the desired state of NicClusterPolicy
 type NicClusterPolicySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	OFEDDriver             *OFEDDriverSpec       `json:"ofedDriver,omitempty"`
-	NVPeerDriver           *NVPeerDriverSpec     `json:"nvPeerDriver,omitempty"`
-	RdmaSharedDevicePlugin *DevicePluginSpec     `json:"rdmaSharedDevicePlugin,omitempty"`
-	SriovDevicePlugin      *DevicePluginSpec     `json:"sriovDevicePlugin,omitempty"`
-	SecondaryNetwork       *SecondaryNetworkSpec `json:"secondaryNetwork,omitempty"`
+	ClusterNodeAffinity     *v1.NodeAffinity `json:"nodeAffinity,omitempty"`
+	ClusterNodeSelectorSpec `json:""`
```

Comment:
no need to specify its field ? `nodeSelector` in the json annotation ?

Reply:
As the struct is just a wrapper over `map[string]string` that represents the nodeSelector, it doesn't map to any json field directly. `nodeSelector` annotation is defined for the struct member.

Reply:
ah thats embedded struct, missed that :)